hardware_interface/test/test_register_actuators.cpp

```diff
@@ -0,0 +1,125 @@
+    {
+      return HW_RET_OK;
+    }
+
+    hw::hardware_interface_ret_t read() override
+    {
+      return HW_RET_OK;
+    }
+
+    hw::hardware_interface_ret_t write() override
+    {
+      return HW_RET_OK;
+    }
+  };
+
+public:
+  TestActuators()
+  {
+  }
+
+protected:
+  DummyRobotHardware robot_hw;
+};
+
+TEST_F(TestActuators, no_actuators_registered_return_empty_on_all_fronts)
+{
+  EXPECT_THAT(robot_hw.get_registered_actuators(), IsEmpty());
+  EXPECT_THAT(robot_hw.get_registered_actuator_names(), IsEmpty());
+
+  hw::ActuatorHandle handle{"", ""};
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.get_actuator_handle(handle));
+}
+
+TEST_F(TestActuators, can_register_actuator_interfaces)
+{
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, FOO_INTERFACE));
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, BAR_INTERFACE));
+}
+
+TEST_F(TestActuators, can_not_double_register_actuator_interfaces)
+{
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, FOO_INTERFACE));
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, BAR_INTERFACE));
+
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.register_actuator(ACTUATOR_NAME, FOO_INTERFACE));
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.register_actuator(ACTUATOR_NAME, BAR_INTERFACE));
+}
+
+TEST_F(TestActuators, can_not_register_with_empty_fields)
+{
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.register_actuator("", ""));
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.register_actuator(ACTUATOR_NAME, ""));
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.register_actuator("", FOO_INTERFACE));
+}
+
+TEST_F(TestActuators, can_not_get_non_registered_actuators_or_interfaces)
+{
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, FOO_INTERFACE));
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR2_NAME, BAR_INTERFACE));
+
+  hw::ActuatorHandle handle1{ACTUATOR_NAME, BAR_INTERFACE};
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.get_actuator_handle(handle1));
+  hw::ActuatorHandle handle2{ACTUATOR2_NAME, FOO_INTERFACE};
+  EXPECT_EQ(HW_RET_ERROR, robot_hw.get_actuator_handle(handle2));
+}
+
+TEST_F(TestActuators, can_get_registered_actuators)
+{
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, FOO_INTERFACE));
+  EXPECT_THAT(robot_hw.get_registered_actuator_names(), ElementsAre(ACTUATOR_NAME));
+  const auto registered_actuators = robot_hw.get_registered_actuators();
+  EXPECT_THAT(registered_actuators, SizeIs(1));
+  const auto & actuator_handle = registered_actuators[0];
+  EXPECT_EQ(actuator_handle.get_name(), ACTUATOR_NAME);
+  EXPECT_EQ(actuator_handle.get_interface_name(), FOO_INTERFACE);
+
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR_NAME, BAR_INTERFACE));
+  EXPECT_THAT(robot_hw.get_registered_actuator_names(), ElementsAre(ACTUATOR_NAME));
+
+  EXPECT_EQ(HW_RET_OK, robot_hw.register_actuator(ACTUATOR2_NAME, FOO_INTERFACE));
+  EXPECT_THAT(
+    robot_hw.get_registered_actuator_names(),
+    UnorderedElementsAre(ACTUATOR_NAME, ACTUATOR2_NAME));
+}
```

Comment:
There is no test that gets an actuator handle successfully.

Reply:
added

Reply:
Thanks.